bitrepository-client/src/main/java/org/bitrepository/commandline/PutFileCmd.java

```diff
@@ -158,14 +174,24 @@ private OperationEvent putTheFile() {
         ChecksumDataForFileTYPE validationChecksum = getValidationChecksum();
         ChecksumSpecTYPE requestChecksum = getRequestChecksumSpecOrNull();
 
-        boolean printChecksums = cmdHandler.hasOption(Constants.REQUEST_CHECKSUM_TYPE_ARG);
+        boolean printChecksums = cmdHandler.hasOption(REQUEST_CHECKSUM_TYPE_ARG);
+        String saltedChecksum = null;
+        if (requestChecksum != null && requestChecksum.isSetChecksumSalt()) {
+            try {
+                FileExchange fileExchange = ProtocolComponentFactory.getInstance().getFileExchange(settings);
+                saltedChecksum = ChecksumUtils.generateChecksum(fileExchange.getFile(url), requestChecksum);
+                // TODO: Compare checksum of URL with the cmd-input checksum?
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Could not convert the given URL to URI.");
```

Comment:
I think something like "Could not retrieve file from '" + url + "'" as in ChecksumStorageModel would be a better error message.